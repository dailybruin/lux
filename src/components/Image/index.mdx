---
name: Image
route: /image
---

import { Playground, PropsTable } from 'docz'
import Image from '.'

# Image

An image with a caption.

<PropsTable of={Image} />

<Playground>
  <Image
    url="http://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg"
    caption="Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA menâ€™s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated."
    credit="Amy Dixon/Photo editor"
    alt="Ashworth Molthen playing water polo."
  />
</Playground>
