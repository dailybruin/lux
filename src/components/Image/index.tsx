import * as React from 'react'
import { css } from 'react-emotion'

export interface ImageProps {
  url: string
  caption: string
  credit: string
  alt: string
}

export default function Image(props: ImageProps) {
  if (!props.alt) {
    throw new Error('Image requires a meaningful `alt` value.')
  }

  return (
    <figure>
      <img
        className={css`
          width: 100%;
        `}
        src={props.url}
        alt={props.alt}
      />
      <figcaption>
        {props.caption} ({props.credit})
      </figcaption>
    </figure>
  )
}
