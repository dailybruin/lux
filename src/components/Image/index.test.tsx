import * as React from 'react'
import { cleanup, render } from 'react-testing-library'
import Image from '.'

describe('Image', () => {
  afterEach(cleanup)

  it('Matches its snapshot', () => {
    const { container } = render(
      <Image
        url="http://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg"
        caption="Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA men’s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated."
        credit="Amy Dixon/Photo editor"
        alt="Ashworth Molthen playing water polo."
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Throws an error if it does not receive an alt prop', () => {
    expect(() => {
      render(
        <Image
          url="http://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg"
          caption="Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA men’s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated."
          credit="Amy Dixon/Photo editor"
          alt=""
        />
      )
    }).toThrowError()
  })
})
