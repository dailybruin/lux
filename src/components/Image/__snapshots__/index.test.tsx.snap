// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Matches its snapshot 1`] = `
.emotion-0 {
  width: 100%;
}

<figure>
  <img
    alt="Ashworth Molthen playing water polo."
    class="emotion-0"
    src="http://dailybruin.com/images/2018/10/web.sp_.mwp_.nbk_.ADX_-640x461.jpg"
  />
  <figcaption>
    Freshman attacker Ashworth Molthen has registered 21 goals on the season, finding the back of the net in 13 of the 16 games so far this year for UCLA menâ€™s water polo. UCLA is one of two teams remaining in the MPSF conference that remains undefeated.
     (
    Amy Dixon/Photo editor
    )
  </figcaption>
</figure>
`;
